### PR TITLE
[TEVA-1413] flatten jobs search params

### PR DESCRIPTION
**Pull request by cpjmcquillan**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1413

## Changes in this PR
- Flatten search params
- Use GOVUKDesignSystemFormBuilder for the sort form on the jobs page
- Fix https://rollbar.com/dfe/teacher-vacancies/items/2332/
- Fix https://rollbar.com/dfe/teacher-vacancies/items/2480/
- Fix https://rollbar.com/dfe/teacher-vacancies/items/2481/
- Fix https://rollbar.com/dfe/teacher-vacancies/items/2488/

### CODE REVIEW

File: app/frontend/src/application/hiring_staff/feedbackForm.js [hunk 1]
```diff
@@ -29,26 +29,28 @@
 const FORM_ELEMENT_ERROR_CLASS = 'govuk-input--error';
 
 document.addEventListener('DOMContentLoaded', () => {
-  Array.from(document.getElementsByClassName('vacancy-feedback__form')).forEach((form) => {
-    form.addEventListener('submit', (event) => {
-      Array.from(document.querySelectorAll(`select[form="${event.target.id}"]`)).forEach((select) => {
-        if (select.value === '') {
-          if (!formElementHasError(select)) {
-            formElementAddError(select, event.target.dataset.optionNotSelectedMessage);
+  if (document.querySelector('.expired__vacancy-form')) {
```
Comment (alexbowen) [inline app/frontend/src/application/hiring_staff/feedbackForm.js:32]: is this actually part of this task or is it aa oversight by me on changes i made earlier in week?

**Review by alexbowen — APPROVED** [review]
a question below but it doesnt really impact me approving

File: app/views/vacancies/_sorting_options.html.haml [hunk 2]
```diff
@@ -1,7 +1,16 @@
-= form_tag('', method: :get, id: 'jobs_sort_form') do
-  - search_criteria.each do |filter, value|
-    - unless filter == :jobs_sort
-      = hidden_field_tag filter, value
-  = label_tag 'jobs_sort_select', t('jobs.sort_by.label'), class: "govuk-label inline", id: 'jobs_sort_label'
-  = select_tag 'jobs_sort', options_for_select(job_sorting_options, @vacancies_search.sort_by), class: 'govuk-select govuk-input--width-10', id: 'jobs_sort_select', disabled: @vacancies.none?
-  = submit_tag t('jobs.sort_by.submit'), class: 'govuk-button govuk-!-margin-0 govuk-input--width-5', id: 'submit_job_sort', disabled: @vacancies.none?
+= form_for @jobs_search_form, as: "", url: jobs_path, method: :get, id: "jobs_sort_form" do |f|
```
Comment (cesidio) [inline app/views/vacancies/_sorting_options.html.haml:1]: 👍  very nice!

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -50,10 +50,13 @@ def current_session_id
   end
 
   def strip_empty_checkboxes(form_key, fields)
-    if params[form_key].present?
-      fields.each do |field|
-        params[form_key][field] = params[form_key][field]&.reject(&:blank?) unless params[form_key][field].is_a?(String)
-      end
+    params_to_strip = if form_key.present? && params[form_key].present?
```
Comment (cesidio) [inline app/controllers/application_controller.rb:53]: shouldn't checking that `params[form_key].present?` is present  to be enough or you need to check also for form_key.present?

**Review by cesidio — APPROVED** [review]

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (cpjmcquillan) [reply app/controllers/application_controller.rb:53]: It was a bit lazy from me, let me do something better

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -49,11 +49,14 @@ def current_session_id
     session.to_h['session_id']
   end
 
-  def strip_empty_checkboxes(form_key, fields)
-    if params[form_key].present?
-      fields.each do |field|
-        params[form_key][field] = params[form_key][field]&.reject(&:blank?) unless params[form_key][field].is_a?(String)
-      end
+  def strip_empty_checkboxes(fields, form_key = nil)
+    params_to_strip = if params[form_key].present?
```
Comment (cesidio) [inline app/controllers/application_controller.rb:53]: one-liner is more readable I believe

Comment (cpjmcquillan) [reply app/controllers/application_controller.rb:53]: Rubocop keeps doing this style 😔🤖

Comment (cesidio) [reply app/controllers/application_controller.rb:53]: if you type this:

```
params_to_strip = params[form_key].present? ? params[form_key] : params
```

rubocop complains?

Comment (cpjmcquillan) [reply app/controllers/application_controller.rb:53]: Ah no I see what you mean now. Yeah I prefer this too 👍🏼

**Pull request merged**